spec/lib/tasks/publishing_api_spec.rb

```diff
@@ -0,0 +1,106 @@
+require "rails_helper"
+
+RSpec.describe "publishing_api rake tasks", type: :task do
+  describe "publishing_api:publish_finders" do
+    before(:each) do
+      Rake::Task["publishing_api:publish_finders"].reenable
+      stub_any_publishing_api_put_content
+      stub_any_publishing_api_patch_links
+      stub_any_publishing_api_publish
+    end
+
+    it "publishes all finders to the Publishing API" do
+      allow(Rails.application.config).to receive(:publish_pre_production_finders).and_return(true)
+      finders = Dir.glob("lib/documents/schemas/*.json")
+      content_ids = finders.map do |json_schema|
+        MultiJson.load(File.read(json_schema))["content_id"]
+      end
+
+      expect { Rake::Task["publishing_api:publish_finders"].invoke }.to output.to_stdout
+
+      content_ids.each do |content_id|
+        assert_publishing_api_put_content(content_id)
+        assert_publishing_api_patch_links(content_id)
+        assert_publishing_api_publish(content_id)
+      end
+    end
+
+    context "an error triggers the rescue state" do
+      it "returns an error message" do
+        stub_any_publishing_api_put_content.and_raise(GdsApi::HTTPServerError.new(500))
+        error_message = %r{Error publishing finder: #<GdsApi::HTTPServerError: GdsApi::HTTPServerError>}
+
+        expect { Rake::Task["publishing_api:publish_finders"].invoke }.to output(error_message).to_stdout
+      end
+    end
+  end
+
+  describe "publishing_api:publish_finder" do
+    before(:each) do
+      Rake::Task["publishing_api:publish_finder"].reenable
+      stub_any_publishing_api_put_content
+      stub_any_publishing_api_patch_links
+      stub_any_publishing_api_publish
+    end
+
+    context "incorrect file name is given" do
+      it "returns an error message" do
+        error_message = %r{#<RuntimeError: Could not find file: lib/documents/schemas/aaib_reportings.json>}
+
+        expect { Rake::Task["publishing_api:publish_finder"].invoke("aaib_reportings") }.to output(error_message).to_stdout
+      end
+    end
+
+    context "correct file name given" do
+      context "no server error present" do
+        it "publishes a single finder to the Publishing API" do
+          content_id = "b7574bba-969f-4c49-855a-ae1586258ff6"
```

Comment:
Do we ever expect this to change? I'm assuming not, but if there's a chance it will we could consider grabbing it from the schema like you did in the above test 

```
schema = Dir.glob("lib/documents/schemas/aaib_reports.json")
content_id =   MultiJson.load(File.read(schema))["content_id"]
```

Please feel free to ignore if irrelevant!